update requestId and userAgent in RequestMessage with RequestOptions

rdtr opened the pull request:
a new PR for https://github.com/apache/tinkerpop/pull/1226

We miss to put requestId and userAgent in bytecode query submission while they are available in requestOptions. This commits fix the issue.

I ran the tests locally and it worked fine. Please let me know if the change any issue. Thanks.

spmallette commented on gremlin-server/src/test/java/org/apache/tinkerpop/gremlin/server/GremlinDriverIntegrateTest.java:
```diff
@@ -64,13 +64,7 @@
-import java.util.Iterator;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
thanks for the test. it looks like your IDE auto-formatted the imports. please note that we don't use wildcard imports in our code style. could you please revert that change?

rdtr replied:
Ah you're right, when I added `import java.util.UUID;`, IntelliJ automatically squashed the imports. Reverted it. Thanks !

spmallette commented:
Other than the wildcard import and a CHANGELOG entry, I think this PR looks good:

VOTE +1

divijvaidya commented:
VOTE +1

rdtr commented:
@spmallette @divijvaidya Thanks for helping me to move this PR forward !

I restored the squashed imports and added a comment to CHANGELOG.asciidoc.
For some reason my addition to the CHANGELOG is detected as a conflict.

Could you resolve the conflict and merge this if the comment I added looks good ?

rdtr commented on CHANGELOG.asciidoc:
```diff
@@ -28,7 +28,10 @@ This release also includes changes from <<release-3-3-10, 3.3.10>>.
 * Expanded the use of `by(String)` modulator so that it can work on `Map` as well as `Element`.
 * Improved error messaging for `by(String)` so that it is more clear as to what the problem is
 * Bump to Netty 4.1.42
```
The last line is the one I want to add in CHANGELOG, but GitHub complains that there is a conflict and I had to add these 3 lines above (GitHub still complains though). I am not clear what should be the right result here.

Please let me know if I need to modify this part.

spmallette replied:
I don't think your PR should contain any lines beyond the last one in this diff. I would just revert your CHANGELOG modification in your branch, rebase the whole PR on tp34 and then add your one line back (and squash) so that it's clean. does that make sense? 

Worst case, if it's taking too much time and you can't figure it out, just revert your CHANGELOG modification, copy/paste your CHANGELOG entry to the comments on this PR and I'll add it manually on merge. thanks.

rdtr replied:
@spmallette Ok, this happened because my `tp34` is behind from `upstream`. I updated `tp34` and rebased this PR onto it, now the conflict is gone. It should be good now.

rdtr commented:
Not this branch should be good to merge. Let me know if further tweaks are needed.
